services/motion/builtin/move_request.go

```diff
@@ -632,6 +632,9 @@ func (ms *builtIn) relativeMoveRequestFromAbsolute(
 	worldObstacles []spatialmath.Geometry,
 	valExtra validatedExtra,
 ) (*moveRequest, error) {
+	if spatialmath.PoseAlmostEqualEps(goalPoseInWorld, spatialmath.NewZeroPose(), motionCfg.planDeviationMM) {
```

Comment:
A couple thoughts here:

- PoseAlmostEqualEps looks at orientation, not just position, so planDeviationMM is not necessarily correct here
- A sufficiently different orientation will cause the above to plan even if located directly on the goal site, even in position-only mode
- Most important: goalPoseInWorld has not yet been placed relative to where the base is, so while comparing it to the zero pose is fine for MoveOnGlobe, I don't think this is correct for MoveOnMap; if the base is located somewhere other than the map origin, then a coincident pose would put goalPoseInWorld at that pose, not at the zero pose. The pose that should be compared to the zero pose is `goal`, about 25 lines further down than this.

Reply:
Also to add onto the third bullet point; this means that it becomes impossible to plan to the origin of a SLAM map.

Reply:
great catch! I'll update this right now.